runtime/j9vm/jvm.c

```diff
@@ -3991,10 +3995,35 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 		JavaVM *vm = (JavaVM *)javaVM;
 		(*vm)->GetEnv(vm, (void **)&env, JNI_VERSION_1_2);
 		if (NULL != env) {
-			char errMsg[512];
-			jio_snprintf(errMsg, sizeof(errMsg), "Failed to load library \"%s\"", libName);
-			errMsg[sizeof(errMsg) - 1] = '\0';
+			char errBuf[512];
+			char *errMsg = errBuf;
+			int bufSize = sizeof(errBuf);
+			const char *portMsg = attemptedLoad ? j9error_last_error_message() : "";
+			if (0 != *portMsg) {
+				/* The port library message includes the file name which failed to load. */
+				bufSize = jio_snprintf(errMsg, bufSize, "Failed to load library %s", portMsg);
+			} else {
+				bufSize = jio_snprintf(errMsg, bufSize, "Failed to load library \"%s\"", libName);
+			}
+			errBuf[sizeof(errBuf) - 1] = '\0';
+			bufSize += 1;
+			if (bufSize > sizeof(errBuf)) {
+				errMsg = (char *)j9mem_allocate_memory(bufSize, OMRMEM_CATEGORY_VM);
+				if (NULL != errMsg) {
+					if (0 != *portMsg) {
+						/* The port library message includes the file name which failed to load. */
+						jio_snprintf(errMsg, bufSize, "Failed to load library %s", portMsg);
+					} else {
+						jio_snprintf(errMsg, bufSize, "Failed to load library \"%s\"", libName);
+					}
+				} else {
+					errMsg = errBuf;
+				}
+			}
 			throwNewUnsatisfiedLinkError(env, errMsg);
+			if (errMsg != errBuf) {
+				j9mem_free_memory(errMsg);
+			}
 		}
 	}
 #endif /* JAVA_SPEC_VERSION >= 17 */
```

Comment:
The condition on line 3992 and here should be `JAVA_SPEC_VERSION >= 11`; see the function description.

Reply:
That's out of scope of my change. It was already JAVA_SPEC_VERSION >= 17 and 11 seems to work fine without it. Searching the OpenJDK 11 code, JVM_LoadLibrary is only used from ClassLoader.c, which isn't used with OpenJ9.

Reply:
Fair enough. Perhaps the function description could be improved to clarify that although the signature changed in jdk11, it is only used by jdk17+.